bus/uploadedsectors.go

```diff
@@ -0,0 +1,60 @@
+package bus
+
+import (
+	"sync"
+	"time"
+
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+)
+
+const cacheExpiry = 12 * time.Hour
+
+type uploadedSectorsCache struct {
+	mu      sync.Mutex
+	uploads map[api.UploadID]time.Time
+	sectors map[api.UploadID]map[types.FileContractID][]types.Hash256
```

Comment:
I considered doing `map[types.Hash256]struct{}` here but using a slice saves me some looping and initialisation at the cost of having potential duplicates in this array, although that really shouldn't be happening

Reply:
I don't think duplicates would cause any issues even if they happened?

But why not introduce a `ongoingUpload` type like
```go
type ongoingUpload struct {
  time.Time
  map[types.FileContractID][]types.Hash256]
}
```
to then do 
```go
uploads map[api.UploadID]*ongoingUpload
```

Reply:
wanted to avoid having a mutex on `ongoingUpload` but reworked it anyway, duplicates are not an issue but I'm annoyed that `/contract/:id/roots` on the `bus` would give such a poor response with little context as to what's actually in the database and what isn't